Userland/Libraries/LibIDL/IDLParser.cpp

```diff
@@ -148,12 +149,14 @@ Optional<Interface&> Parser::resolve_import(auto path)
         report_parsing_error(DeprecatedString::formatted("Circular import detected: {}", include_path), filename, input, lexer.tell());
     import_stack.set(real_path);
 
-    auto file_or_error = Core::File::open(real_path, Core::OpenMode::ReadOnly);
+    auto file_or_error = Core::Stream::File::open(real_path, Core::Stream::OpenMode::Read);
     if (file_or_error.is_error())
         report_parsing_error(DeprecatedString::formatted("Failed to open {}: {}", real_path, file_or_error.error()), filename, input, lexer.tell());
 
-    auto data = file_or_error.value()->read_all();
-    auto& result = Parser(this, real_path, data, import_base_path).parse();
+    auto data_or_error = file_or_error.value()->read_all();
+    if (data_or_error.is_error())
+        report_parsing_error(DeprecatedString::formatted("Failed to read {}: {}", real_path, data_or_error.error()), filename, input, lexer.tell());
+    auto& result = Parser(this, real_path, data_or_error.value(), import_base_path).parse();
```

Comment:
Can we continue if the data is an error? You can't even do `.value()` right?

Reply:
`report_parsing_error` prints the given error and then does `exit(1)`. It looks a bit confusing, but it's in line with what the rest of the code does.

Reply:
Ah didn't see that, my bad!